app.rb

```diff
@@ -0,0 +1,139 @@
+# create a App class
+require_relative 'person'
+require_relative 'book'
+require_relative 'student'
+require_relative 'teacher'
+
+class App
+  def initialize
+    @books = []
+    @person = []
+    @rentals = []
+  end
+
+  # create student method
+  def create_student
+    print "Name : "
+    name = gets.chomp
+    print "Age : "
+    age = gets.chomp
+
+    student = Student.new(age, nil, name)
+    @people.push(student)
+
+    puts "New student created successfully"
+  end
+
+  # create teacher method
+  def create_teacher
+    print "Name : "
+    name = gets.chomp
+    print "Age : "
+    age = gets.chomp
+    print "Specialization : "
+    specialization = gets.chomp
+    teacher = Teacher.new(age, specialization, name)
+    @people.push(teacher)
+    puts "New teacher created successfully"
+  end
+
+  # Create person method
+  def create_person
+    print "Do you want to create a student (1) or a teacher (2)? [Input the number]: "
+    choice = gets.chomp
+    case choice
+    when "1"
+      create_student
+    when "2"
+      create_teacher
+    else
+      puts "Please insert a valid number [1 or 2]"
+      return
+    end
+    puts "Person created successfully"
+  end
+
+  # create book method
+  def create_book
+    print "Title : "
+    title = gets.chomp
+    print "Author : "
+    author = gets.chomp
+
+    # book instance
+    book = Book.new(title, author)
+    @books.push(book)
```

Comment:
[OPTIONAL]

- I think it will be better if you add the notice that book has been created successfully if the book is actually saved 👍🏽